src/modules/peek/Peek.UI/MainWindowViewModel.cs

```diff
@@ -4,16 +4,55 @@
 
 namespace Peek.UI
 {
-    using System.Collections.Generic;
+    using System;
     using CommunityToolkit.Mvvm.ComponentModel;
-    using Peek.Common.Models;
+    using Microsoft.UI.Xaml;
 
     public partial class MainWindowViewModel : ObservableObject
     {
-        [ObservableProperty]
-        private File? currentFile;
+        private const int NavigationThrottleDelayMs = 100;
+
+        public MainWindowViewModel()
+        {
+            navigationThrottleTimer.Tick += NavigationThrottleTimer_Tick;
+            navigationThrottleTimer.Interval = TimeSpan.FromMilliseconds(NavigationThrottleDelayMs);
+        }
+
+        public void AttemptLeftNavigation()
```

Comment:
nit: naming: TryLeftNavigation? maybe it should return false or true based on if the navigation was successful?

Reply:
Yeah I agree - eventually we'll probably want some ui feedback if navigation fails. I'll add some Todos since, we'll also need to update code in the querying class